simSPI/linear_simulator/dataset_generator.py

```diff
@@ -0,0 +1,139 @@
+"""runs dataset generator and saves it."""
+import os
+
+import mrcfile
+import pandas as pd
+import starfile
+import torch
+
+from .modules import simulator
+from .params_utils import ParamsFactory
+from .starfile_utils import return_names, starfile_data
+
+"""Module to generate and save dataset (including metadata) in the output directory."""
+
+
+class dataset_gen(torch.nn.Module):
+    """class to generate and save dataset (including metadata) in the output directory.
+
+    Parameters
+    ----------
+    config: class
+        Class containing parameters of the dataset generation and simulator.
+
+    """
+
+    def __init__(self, config):
+        super(dataset_gen, self).__init__()
+
+        self.config = config
+        self.simulator = simulator(config)  # instantiating the simulator
+        self.dataframe = []
+        self.init_dir()  # initializing the output folder
+        self.print_statements()
+        self.params_generator = ParamsFactory.get_params_generator(
+            config
+        )  # instantiating params generator
+
+    def run(self):
+        """Generate a chunk of projection and save it."""
+        for iterations in range(
+            self.config.datasetSize // self.config.chunks
+        ):  # TODO: last few images will be left
+            rot_params, ctf_params, shift_params = self.params_generator.get_params()
+            projections = self.simulator(rot_params, ctf_params, shift_params)
+            self.save_mrc(projections, iterations)
+            self.starfile_data(rot_params, ctf_params, shift_params, iterations)
+        self.save_starfile()
+        self.save_configfile()
+
```

Comment:
I suggest writing .mrc files of a reasonable size, and breaking up into batches if they get too big. 

As written, won't this overwrite "Simulated.star"?